service/src/main/java/bio/terra/workspace/service/petserviceaccount/PetSaService.java

```diff
@@ -54,35 +53,51 @@ public PetSaService(
    * Wrapper around {@code enablePetServiceAccountImpersonationWithEtag} without requiring a
    * particular GCP eTag value.
    */
-  public Policy enablePetServiceAccountImpersonation(UUID workspaceId, UserWithPetSa userAndPet) {
+  public Policy enablePetServiceAccountImpersonation(
+      UUID workspaceId, AuthenticatedUserRequest userRequest) {
     // enablePetServiceAccountImpersonationWithEtag will only return an empty optional if the
     // provided eTag does not match current policy. Because we do not use eTag checking here, this
     // is always nonempty.
-    return enablePetServiceAccountImpersonationWithEtag(workspaceId, userAndPet, null).get();
+    return enablePetServiceAccountImpersonationWithEtag(workspaceId, userRequest, null).orElseThrow(  () ->
+            new RuntimeException(
+                    "Error enabling user's proxy group to impersonate pet SA"));
   }
 
   /**
-   * Grant a user permission to impersonate their pet service account in a given workspace. Unlike
-   * other operations, this does not run as a flight because it only requires one write operation.
-   * This operation is idempotent.
+   * Grant a user's proxy group permission to impersonate their pet service account in a given
+   * workspace. Unlike other operations, this does not run as a flight because it only requires one
+   * write operation. This operation is idempotent.
    *
-   * <p>This method requires a user's pet service account email as input. As a transitive
-   * dependency, this also means the provided workspace must have a GCP context.
+   * <p>The provided workspace must have a GCP context.
    *
    * <p>This method does not authenticate that the user should have access to impersonate their pet
    * SA, callers should validate this first.
    *
    * @param workspaceId ID of the workspace to enable pet SA in
-   * @param userAndPet The user and the pet SA the user is being granted permission to
+   * @param userRequest The user whose proxy group will be granted permission.
    * @param eTag GCP eTag which must match the pet SA's current policy. If null, this is ignored.
    * @return The new IAM policy on the user's pet service account, or empty if the eTag value
    *     provided is non-null and does not match current IAM policy on the pet SA.
    */
   public Optional<Policy> enablePetServiceAccountImpersonationWithEtag(
-      UUID workspaceId, UserWithPetSa userAndPet, @Nullable String eTag) {
+      UUID workspaceId, AuthenticatedUserRequest userRequest, @Nullable String eTag) {
+    // userRequest might not have user email, so fetch user email explicitly.
+    String userEmail =
```

Comment:
I guess you could try to get the email from the request first, before calling Sam.